### PR TITLE
asmjs: anonymous function crash

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

**Comment by obastemur** [comment]
Also added a test case

**Comment by obastemur** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_release

**Comment by MikeHolman** [comment]
:shipit:

File: lib/Runtime/Language/AsmJsModule.cpp [hunk 1]
```diff
@@ -1080,6 +1080,11 @@ namespace Js
     AsmJsFunc* AsmJsModuleCompiler::CreateNewFunctionEntry( ParseNode* pnodeFnc )
     {
         PropertyName name = ParserWrapper::FunctionName( pnodeFnc );
+        if ( !name )
```
Comment (ThomsonTan) [inline lib/Runtime/Language/AsmJsModule.cpp]: >name [](start = 14, length = 4)

Do we need use a default name like 'anonymous function' for it?

Comment (Cellule) [inline lib/Runtime/Language/AsmJsModule.cpp]: No, this should be invalid asm.js. Returning null here will invalidate the asm.js module and then be parsed as normal javascript

**Comment by obastemur** [comment]
@Cellule @MikeHolman @ThomsonTan Thanks for the reviews!

**Pull request merged**